lib/xnvme_spec.c

```diff
@@ -748,6 +748,148 @@ xnvme_spec_nvm_idfy_ns_pr(struct xnvme_spec_nvm_idfy_ns *idfy, int opts)
 	return xnvme_spec_nvm_idfy_ns_fpr(stdout, idfy, opts);
 }
 
+int
+xnvme_spec_drecv_idfy_fpr(FILE *stream, struct xnvme_spec_idfy_dir_rp *idfy, int opts)
+{
+	int wrtn = 0;
+
+	switch (opts) {
+	case XNVME_PR_DEF:
+	case XNVME_PR_YAML:
+		break;
+
+	case XNVME_PR_TERSE:
+		wrtn += fprintf(stream, "# ENOSYS: opts(%x)", opts);
+		return wrtn;
+	}
+
+	wrtn += fprintf(stream, "xnvme_spec_idfy_dir_rp:");
+	if (!idfy) {
+		wrtn += fprintf(stream, " ~\n");
+		return wrtn;
+	}
+
+	wrtn += fprintf(stream, "\n");
+	wrtn += fprintf(stream, "  Directives supported:\n");
+	wrtn += fprintf(stream, "    identify: %d\n", idfy->directives_supported.identify);
+	wrtn += fprintf(stream, "    streams: %d\n", idfy->directives_supported.streams);
+	wrtn += fprintf(stream, "  Directives enabled:\n");
```

Comment:
This should be printed in YAML, something like:
```
  directives_enabled:\n
```

Reply:
ok, done thanks